### PR TITLE
Fb content security policy

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: api/src/org/labkey/api/security/AuthenticatedResponse.java [hunk 1]
```diff
@@ -39,15 +46,164 @@ public AuthenticatedResponse(HttpServletResponse response)
...
+//        // for debugging you can do this
+//        if (_safeWriter == null)
+//            _safeWriter = new SafePrintWriter(super.getWriter());
+//        return _safeWriter;
+    }
+
+    @Override
+    public void addHeader(String name, String value)
+    {
+        if (StringUtils.equalsAnyIgnoreCase("Content-Security-Policy"))
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/security/AuthenticatedResponse.java]: Any reason you're not using a simple .equalsIgnoreCase() here? This code isn't using the name parameter at all so it'll never fire the exception

File: api/src/org/labkey/api/security/AuthenticatedResponse.java [hunk 2]
```diff
@@ -39,15 +46,164 @@ public AuthenticatedResponse(HttpServletResponse response)
...
+        StringBuilder content = new StringBuilder();
+        var semi = "";
+        for (var e : _contentSecurityPolicy.asMap().entrySet())
+        {
+            content.append(semi).append(e.getKey());
+            semi = "; ";
+            for (String s : e.getValue())
+                content.append(" ").append(s);
+        }
+        super.setHeader("Content-Security-Policy", content.toString());
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/security/AuthenticatedResponse.java]: There's enough instances of this string to extract a constant

File: study/src/org/labkey/study/controllers/StudyController.java [hunk 3]
```diff
@@ -4688,6 +4692,8 @@ public ModelAndView getView(CustomizeParticipantViewForm form, boolean reshow, B
             {
                 // issue : 18595 chrome will complain that the script we are executing matches the script sent down in the request
                 getViewContext().getResponse().setHeader("X-XSS-Protection", "0");
```
Comment (labkey-jeckels) [inline study/src/org/labkey/study/controllers/StudyController.java]: Can we delete this old header usage too?

File: internal/src/org/labkey/api/security/AuthFilter.java [hunk 4]
```diff
@@ -80,7 +87,6 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         {
             if (!"ALLOW".equals(AppProps.getInstance().getXFrameOptions()))
                 resp.setHeader("X-Frame-Options", AppProps.getInstance().getXFrameOptions());
-            resp.setHeader("X-XSS-Protection", "1; mode=block");
```
Comment (labkey-jeckels) [inline internal/src/org/labkey/api/security/AuthFilter.java:83]: Do we not need to migrate this one too?

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]

File: api/src/org/labkey/api/security/AuthenticatedResponse.java (hunk 1 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/security/AuthenticatedResponse.java]: oops.

Still working on it (hence draft). Just thought you'd be curious.

**Pull request closed without merging**